File: javascript/6_kyu/build_tower.js
// Build Tower
// https://www.codewars.com/kata/576757b1df89ecf5bd00073b/train/javascript

function towerBuilder(n) {
    if ( n == 1) {
        return ["*"];
    } else {
        var length = n + (n - 1);
        var ans = [];
        
        for (var i = 1; i < length + 1; i += 2) {
            var spaces = length - i;
            var str1 = ' '.repeat(parseInt(spaces / 2));
            var str2 = str1.concat("*".repeat(i));
            var str3 = str2.concat(' '.repeat(parseInt(spaces / 2)));
            ans.push(str3);
        }
        return ans;
    }
}

console.log(towerBuilder(1));
console.log(towerBuilder(2));
console.log(towerBuilder(3));

File: python/6_kyu/build_tower.py
# Build Tower
# https://www.codewars.com/kata/576757b1df89ecf5bd00073b/train/python

# Solution 1: Concatenation
def tower_builder(n):
    if n == 1:
        return ['*']
    else:
        length = n + (n - 1)
        ans = []

        for star in range(1, length + 1, 2):
            spaces = length - star
            sep = ' ' * int(spaces / 2)
            stars = '*' * star
            ans.append(sep + stars + sep)

        return ans

print(tower_builder(1))
print(tower_builder(2))
print(tower_builder(3))
